Application gateway plugin crash

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.3.7

### AzureRM Provider Version

3.45.0

### Affected Resource(s)/Data Source(s)

azurerm_application_gateway

### Terraform Configuration Files

```hcl
terraform {
  required_version = "> 1.0.0"
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "> 3.0.0"
    }
  }
  backend "azurerm" {
    resource_group_name  = "infra-htecrnd-terraform-rg-eastus2"
    storage_account_name = "infrahtecrndshared"
    container_name       = "xxxxx"
    key                  = "xxxxxx"
  }
}

provider "azurerm" {
  features {}
}
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.45.0_x5.exe plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 32 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.expandApplicationGatewayURLPathMaps(0x6debfac?, {0xc00015a210, 0xac})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/network/application_gateway_resource.go:4309 +0x1765
        github.com/hashicorp/terraform-provider-azurerm/internal/services/network/application_gateway_resource.go:1579 +0x85c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x749cf80?, {0x749cf80?, 0xc0021b47b0?}, 0xd?, {0x6201100?, 0xc001480c00?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001314a80, {0x749cf80, 0xc0021b47b0}, 0xc002256000, 0xc000312400, {0x6201100, 0xc001480c00})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000397758, {0x749cf80?, 0xc0021b4690?}, 0xc001ed5d10)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001617680, {0x749cf80?, 0xc00218de60?}, 0xc00219e690)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x68dbd60?, 0xc001617680}, {0x749cf80, 0xc00218de60}, 0xc00219e620, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000028000, {0x74ad080, 0xc000ef8000}, 0xc0021a18c0, 0xc001518270, 0xb75dc60, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc000028000, {0x74ad080, 0xc000ef8000}, 0xc0021a18c0, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.45.0_x5.exe plugin crashed!
```


### Expected Behaviour

It should have created application gateway.

### Actual Behaviour

Immediately crashed.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/network/application_gateway_resource.go
+++ b/internal/services/network/application_gateway_resource.go
@@ -4306,7 +4306,10 @@ func expandApplicationGatewayURLPathMaps(d *pluginsdk.ResourceData, gatewayID st
 
 			rulePaths := make([]string, 0)
 			for _, rulePath := range ruleConfigMap["paths"].([]interface{}) {
-				rulePaths = append(rulePaths, rulePath.(string))
+				p, ok := rulePath.(string)
+				if ok {
+					rulePaths = append(rulePaths, p)
+				}
 			}
 
 			rule := network.ApplicationGatewayPathRule{

